CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+project(facil.io C)
+
+find_package(Threads REQUIRED)
+
+set(facil.io_SOURCES
```

Comment:
Cool!

Any way to make this list dynamic using src folders?

I think this will reduce the risk of future changes breaking the linkage.